### PR TITLE
[SPARK-35995][INFRA][3.2][FOLLOWUP] Fix the reference branches and PR builder

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fixes the reference branches and PR builder.

### Why are the changes needed?

Currently the build and tests are on `master` branch because the reference branch is always `master`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by ueshin** [comment]
Hmm, I can't trigger "Notify test workflow" .. ?

**Comment by dongjoon-hyun** [comment]
Thank you for making a PR, @ueshin .

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -3,15 +3,7 @@ name: Build and test
...
-    - branch-3.2
-  pull_request:
-    branches:
-    - branch-3.2
-  workflow_dispatch:
-    inputs:
-      target:
-        description: 'Target branch to run'
-        required: true
+      - '**'
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:6]: Well, is this correct?

The original one is the same with what we did in `branch-3.1`. Do you mean that `branch-3.1` is not working too?
- https://github.com/apache/spark/blob/branch-3.1/.github/workflows/build_and_test.yml

**Comment by dongjoon-hyun** [comment]
If this works correctly, I'm good, @ueshin .

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (ueshin) [reply .github/workflows/build_and_test.yml:6]: I think `branch-3.1` is fine. We changed to distribute the tests to each forked repository after `3.1`, IIRC.

**Comment by dongjoon-hyun** [comment]
Please let me know if the PR is ready. 
- The first commit seems to fail for some reason.
- The second and 3rd commit seems to skip the real tests

**Comment by dongjoon-hyun** [comment]
For the following, it seems that we need @HyukjinKwon . :)
> Hmm, I can't trigger "Notify test workflow" .. ?

**Comment by ueshin** [comment]
> - The second and 3rd commit seems to skip the real tests

The tests are running in my repository. e.g., https://github.com/ueshin/apache-spark/actions/runs/994797276 for the commit https://github.com/apache/spark/pull/33197/commits/e07ac5691677b478db55c12060ce15970492b1fe.

But it doesn't link to this PR because "Notify test workflow" is not working..

> For the following, it seems that we need @HyukjinKwon . :)

Yeah, it would be really helpful!

**Comment by dongjoon-hyun** [comment]
I checked the checkout and merge works correctly .
- https://github.com/ueshin/apache-spark/actions/runs/994797276

The notification seems to weird because it's almost empty.
- https://github.com/ueshin/apache-spark/actions/workflows/notify_test_workflow.yml

For an experiment, could you please make a PR to your branch-3.2 instead of Apache Spark, @ueshin ? In that case, you should have the log.

**Comment by dongjoon-hyun** [comment]
If it works in your repo, I believe we had better merge this PR and see the behavior. This PR is working correctly on branch commits already.

**Comment by ueshin** [comment]
FYI: https://github.com/ueshin/apache-spark/pull/6

**Comment by dongjoon-hyun** [comment]
Ya, I saw it. And, it seems to be triggered at least.
Screen Shot 2021-07-02 at 2 37 01 PM

**Comment by dongjoon-hyun** [comment]
To recover branch-3.2, shall we land this, @ueshin ?

**Review by dongjoon-hyun — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by ueshin** [comment]
Sure, I'm fine!

**Comment by dongjoon-hyun** [comment]
Thank you so much, @ueshin and @viirya .
I did desperately need this test coverage. :)
Merged to branch-3.2.

**Comment by ueshin** [comment]
Thanks!

**Comment by viirya** [comment]
Thanks @ueshin @dongjoon-hyun

**Comment by HyukjinKwon** [comment]
Thanks Takuya! I was about to take a look yesterday but was stuck in some personal works ...

**Pull request closed without merging**